packages/context/src/interceptor-chain.ts

```diff
@@ -8,46 +8,57 @@ import {BindingFilter} from './binding-filter';
 import {BindingAddress} from './binding-key';
 import {BindingComparator} from './binding-sorter';
 import {Context} from './context';
-import {InvocationResult, Next} from './invocation';
+import {InvocationResult} from './invocation';
 import {transformValueOrPromise, ValueOrPromise} from './value-promise';
-
 const debug = debugFactory('loopback:context:interceptor-chain');
 
+/**
+ * The `next` function that can be used to  invoke next generic interceptor in
+ * the chain
+ *
+ * @typeParam T - Return type of `next()`
+ */
+export type Next<T = InvocationResult> = () => ValueOrPromise<T>;
+
 /**
  * An interceptor function to be invoked in a chain for the given context.
  * It serves as the base interface for various types of interceptors, such
  * as method invocation interceptor or request/response processing interceptor.
  *
  * @typeParam C - `Context` class or a subclass of `Context`
+ * @typeParam T - Return type of `next()`
  * @param context - Context object
  * @param next - A function to proceed with downstream interceptors or the
  * target operation
  *
  * @returns The invocation result as a value (sync) or promise (async)
  */
-export type GenericInterceptor<C extends Context = Context> = (
-  context: C,
-  next: Next,
-) => ValueOrPromise<InvocationResult>;
+export type GenericInterceptor<
+  C extends Context = Context,
+  T = InvocationResult
+> = (context: C, next: Next<T | undefined>) => ValueOrPromise<T | undefined>;
```

Comment:
Why do we need `| undefined` part? IMO, it should be up to the user of `GenericInterceptor` to decide if interceptors are allowed to return `undefined`, and they should have the option to force all interceptors to return a defined value.

Reply:
`undefined` can happen when there are no interceptors in the chain.